api/v1/stage_types.go

```diff
@@ -56,7 +56,9 @@ type StageSpec struct {
 	Source Source `json:"source"`
 
 	// Namespace where the application will be deployed.
-	Namespace string `json:"namespace,omitempty"`
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
```

Comment:
Does this force our Portal UI to set this value? Or can we still rely on the operator to populate it for us in case it is empty?

Reply:
Yes, it forces us to set the value.
But in portal UI, it is already required.
![image](https://github.com/user-attachments/assets/64a01fe1-b4b3-480e-b686-593f0f37c646)